solutions/devsprint-matheus-reis-5/FinanceApp/Modules/Transfers/TransfersViewController.swift

```diff
@@ -40,7 +50,20 @@ extension TransfersViewController: TransferViewDelegate {
 }
 
 extension TransfersViewController: ContactListViewControllerDelegate {
+    func selected(contact: Contact) {
+        didSelectContact()
+        selectedContact.accept(contact)
+    }
+}
 
+private extension TransfersViewController {
+    func bindUi() {
+        selectedContact.map { contact in
+            return contact == nil ? "Choose Contact" : contact?.name
+        }.bind(to: transferView.chooseContactButton.rx.title())
```

Comment:
@reisdev  como eu faria para usar o `.bind(to:)` ligando a um componente da camada de view, colocando essa parte de map em uma View Model?

Reply:
Supondo que você tenha a seguinte View Model:

```swift
final class TransfersViewModel {
    private let selectedContact = BehaviorRelay<Contact?>(value: nil)
}
```

Você poderia fazer o bind da seguinte forma:

```swift
viewModel.selectedContact.map { contact in
    return contact?.name ?? "Choose Contact"
}.bind(to: transferView.chooseContactButton.rx.title())
```

Inclusive, sugestão de alteração no operador ternário. Você poderia usar o operador de coalescência nula, que seria o `??`. Como você tem um encadeamento opcional, `contact?.name`, caso o `contact` seja nulo, o map irá retornar `"Choose Contact"`.

Reply:
legal demais, refatorei pra passar esse map pra VM